sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkOperationUtils.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.thriftserver
+
+import org.apache.hive.service.cli.session.HiveSession
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.{SparkSession, SQLContext}
+import org.apache.spark.sql.catalyst.catalog.CatalogTableType
+import org.apache.spark.sql.catalyst.catalog.CatalogTableType.{EXTERNAL, MANAGED, VIEW}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.util.Utils
+
+/**
+ * Utils for Spark operations.
+ */
+private[hive] trait SparkOperationUtils {
```

Comment:
how about
```
trait SparkOperationUtils { this: Operation =>
  ...
  
  override def run(): Unit = {
    withLocalProperties {
      .run()
    }
  }
}
```

You can try this in scala shell
```
class AA {
  def f(): Int = 1
}

trait BB { this: AA =>
  override def f(): Int = 2
}

class CC extends AA with BB

new CC().f // print 2
```

Reply:
Nice. Do you think I could roll the close() function like that as well in this PR? It's repetitive across all operations, only SparkExecuteStatementOperation needs it slightly different.

Reply:
I validated that it works also if there are classes "in the middle" as in this case:
```
class AA {
  def f(): Int = 1
}

trait BB extends AA {
  abstract override def f(): Int = super.f() + 1
}

class CC extends AA {
  override def f(): Int = 4
}

class DD extends CC with BB

new DD().f // prints 5
```